### ISSUE TITLE
Filtering by 'Qual' on cancer variants page

### ISSUE THREAD

**Issue by cogrho** [Clinical-Genomics/scout#4510]
Hi!
Is it possible to filter the cancer variants by 'Qual' to, for example, obtain only 'PASS' variants or filter out cancer-germline variants? Alternatively, can the 'Qual' field be sorted so that 'PASS' variants appear at the top of the page?

Thanks!

**Comment by northwestwitch** [comment]
I'm not really sure how to achieve this, since often VCF files are merged from different VCF files created by dfferent callers, and some callers might have quality pass and some other caller no.. 🤔

**Comment by dnil** [comment]
Each variant line would normally have a single "FILTER" statement (say e.g. PASS or LowQualityFiltered), and this is what is also reflected in the PASS tags we display per variant. 

I'm not sure what your pipeline or location is, or use case, but the way we usually do it with most pipelines we run in Stockholm is the VCFs are hard filtered to a quality status that the analysis is comfortable with based on pipeline validation. Variants also receive a bonus/malus in the ranking system, so that lower quality variants within the soft bounds are found slightly lower than high quality ones, all other things equal. Variants are then triaged as if "true" initially, based on consequence, rank, phenotype etc, and at a point during that triage a mix of quality indicators, e.g. GQ values from the variant format field together with other information like allele depth, total locus depth, region call quality (e.g. is this in a segmental duplication region), the IGV picture (is this a noisy region?) etc. There seems to be little use in loading variants that would never be eligible to be considered true, and conversely to treat them separately?

Please explain your use case a bit and we'll see what we can do! I understand this was intended for some cancer pipeline; if so which one?

**Comment by cogrho** [comment]
In our clinics, we have a custom somatic variant calling pipeline where we merge VCF files from three callers and set FILTER='PASS' in the merged VCF file for variants identified by two or more callers. Additionally, our merged VCF file includes 'germline' filter and filters for low-quality variants. Sometimes, variants flagged as low-quality prove to be true positives upon further inspection, so we prefer soft filtering. Therefore, it would be great to have an option to display only those variants with a green 'PASS' tag in the 'Qual' column, while hiding variants with a yellow 'GERM' tag. This would make the filtering process more transparent for us.

**Comment by dnil** [comment]
Thank you for the details!

We will have a look, it should be very possible to add a qual filter option - perhaps a checkbox.

A couple of points though for why we have chosen a different route. 

First, strongly consider ranking variants. We completely agree on the soft filtering, within limits. If something is just bad, it is. Or too common. Or just too germline, if you are looking at your somatic variants list, and have a germline analysis for the same patient just waiting anyway. Consider removing those way off the charts, that could not be clinical. The rest, sure, but rank them as well, mandatory if you are looking at wgs or panels above a few hundred genes, but also useful for small panels sometimes.

Second, the way you describe multiple callers is something that would bring high specificity given that the callers are very different (orthogonal), and mostly bias if they are not. The way to achieve sensitivity in e.g. SV calling is to include multiple callers, and evaluate them on their own merits. Is you/your developer using the "SET" or "FOUND_IN" INFO tags in the VCF so you get nice badges on each variant? Do you use callers we do not yet support? Consider an issue or PR for that!

**Comment by cogrho** [comment]
A checkbox or a dropdown menu would be great, thanks!

Thank you for the suggestions! We sometimes really end up with a lot of variants.. If I may ask, what kind of ranking system do you use for prioritisation?  We also evaluate each caller separately, and we indeed use callers that are not yet supported - Strelka2 and LoFreq.

**Comment by northwestwitch** [comment]
We now have an institute-level soft filtering syste, see [this PR](https://github.com/Clinical-Genomics/scout/pull/5254). Can this solve this specific issue as well?

**Comment by dnil** [comment]
And regarding ranking, we still use a rather simple weighted linear sum model [GENMOD](https://github.com/Clinical-Genomics/genmod), configured for the annotations we have for different pipelines like e.g. https://github.com/Clinical-Genomics/reference-files/blob/master/rare-disease/rank_model/rank_model_-v1.39-.ini. It has served us well, and consistently gives RD variants among the very top ranked. We still don't have a great model for different somatic applications; it's ever pushed back a bit. Our users survive by adding a lot of custom filters - or by hard filters before uploading.

**Comment by dnil** [comment]
Right, yes, if the other callers are working well, we could just add tags for them to make things look prettier. And if not, just submit a few examples and we can try to help fix the loading.

**Comment by dnil** [comment]
We hope the issue has been solved by institute level Soft Filters (see e.g. #5254 and #6204). Let us know if it didn't cover your use cases for some reason! Happy scouting!
